pugjs/types_test.go

```diff
@@ -170,6 +170,69 @@ func TestArray_Pop(t *testing.T) {
 	}
 }
 
+func TestArray_Shift(t *testing.T) {
+
+	tests := []struct {
+		name           string
+		input          *Array
+		expectedArray  *Array
+		expectedResult Object
+	}{
+		{
+			name:           "test shift with string",
+			input:          convert([]string{"something", "test", "somewhere", "whatever", "last"}).(*Array),
+			expectedArray:  convert([]string{"test", "somewhere", "whatever", "last"}).(*Array),
+			expectedResult: String("something"),
+		},
+		{
+			name:           "test shift with int",
+			input:          convert([]int{1, 2, 4, 3}).(*Array),
+			expectedArray:  convert([]int{2, 4, 3}).(*Array),
+			expectedResult: Number(1),
+		},
+	}
+
+	for _, tt := range tests {
+		array := tt.input
+		result := array.Shift()
+		assert.Equal(t, tt.expectedResult, result)
+		assert.Equal(t, tt.expectedArray.items, array.items)
+	}
+}
+
+func TestArray_Unshift(t *testing.T) {
+
+	tests := []struct {
+		name           string
+		array          *Array
+		addElement     Object
+		expectedResult *Array
+		expectedLength int
+	}{
+		{
+			name:           "test unshift with string",
+			array:          convert([]string{"something", "test"}).(*Array),
+			addElement:     String("first"),
+			expectedResult: convert([]string{"first", "something", "test"}).(*Array),
+			expectedLength: 3,
+		},
+		{
+			name:           "test unshift with int",
+			array:          convert([]int{4, 3}).(*Array),
+			addElement:     Number(1),
+			expectedResult: convert([]int{1, 4, 3}).(*Array),
+			expectedLength: 3,
+		},
```

Comment:
Would it be possible to add `Number` to an `Array` containing `String` and if yes, would this be problematic?

Reply:
Added a test case for that. As long as it is a pugjs.Object it works